pkg/prometheus/operator.go

```diff
@@ -949,7 +949,7 @@ func loadBasicAuthSecret(basicAuth *monitoringv1.BasicAuth, s *v1.SecretList) (B
 
 }
 
-func (c *Operator) loadBasicAuthSecrets(mons map[string]*monitoringv1.ServiceMonitor, remoteReads []monitoringv1.RemoteReadSpec, remoteWrites []monitoringv1.RemoteWriteSpec, s *v1.SecretList) (map[string]BasicAuthCredentials, error) {
+func (c *Operator) loadBasicAuthSecrets(mons map[string]*monitoringv1.ServiceMonitor, remoteReads []monitoringv1.RemoteReadSpec, remoteWrites []monitoringv1.RemoteWriteSpec, apiserverConfig *monitoringv1.APIServerConfig, s *v1.SecretList) (map[string]BasicAuthCredentials, error) {
```

Comment:
There is a pull request pending that fixes some behavior here, we should make sure that this still works as expected for both cases. Could you coordinate across the PRs @mxinden ?

Reply:
Cross referencing: https://github.com/coreos/prometheus-operator/pull/1550

Reply:
update cross referencing: https://github.com/coreos/prometheus-operator/pull/1619